### PR TITLE
FAQs

### PR DESCRIPTION
**Pull request by austinthao5**
Thanks for contributing to the Harness Developer Hub! Our code owners will review your submission.

## Description

* Please describe your changes: \__________________________________
* Jira/GitHub Issue numbers (if any): \______________________________
* Preview links/images (Internal contributors only): \__________________

## PR lifecycle

We aim to merge PRs within one week or less, but delays happen sometimes.

If your PR is open longer than two weeks without any human activity, please tag a [code owner](https://github.com/harness/developer-hub/blob/main/.github/CODEOWNERS) in a comment.

PRs must meet these requirements to be merged:

- [ ] Successful preview build.
- [ ] Code owner review.
- [ ] No merge conflicts.
- [ ] Release notes/new features docs: Feature/version released to at least one prod environment.

### CODE REVIEW

File: kb/continuous-integration/continuous-integration-faqs.md [hunk 1]
```diff
@@ -1027,9 +1027,9 @@ For troubleshooting information for Git event (webhook) triggers, go to [Trouble
 
 ## Pipeline initialization and Harness CI images
 
-### Initialize step to fails with a "Null value" error
+### Initialize step fails with a "Null value" error or timeout
```
Comment (aimurphy) [inline kb/continuous-integration/continuous-integration-faqs.md:1030]: The provided question was a near duplicate of this existing one. I combined them and updated a reference to this question that was elsewhere in the docs.

**Comment by aimurphy** [comment]
approved from CI perspective. Please wait for approval from other writers before merging.

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 2]
```diff
@@ -6872,3 +6872,10 @@ Users can utilize the following API to retry a failed stage: [retryPipeline](htt
 #### When attempting to import a **.yaml file** from GitHub to create a new pipeline, the message `This file is already in use by another pipeline` is displayed. Given that there are no other pipelines in this project, is there a possibility of a duplicate entry that I may not be aware of?
 
 It's possible that there are two pipeline entities in the database, each linked to the same file path from the Harness account and the GitHub URL. Trying to import the file again may trigger the `File Already Imported` pop-up on the screen. However, users can choose to bypass this check by clicking the `Import` button again.
+
+#### Why is my GitOps Sync Step Failing with the error Application does not correspond to the service(s) selected in the pipeline execution?
```
Comment (SushrutHarness) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6876]:
```suggestion
#### Why is my GitOps Sync Step Failing with the error `Application does not correspond to the service(s) selected in the pipeline execution`?
```

**Review by SushrutHarness — COMMENTED** [review]
Left a typo comment

File: kb/continuous-delivery/continuous-delivery-faqs.md (hunk 2 above)
Comment (brian-f-harness) [reply kb/continuous-delivery/continuous-delivery-faqs.md:6876]: @SushrutHarness For other modules and PL, we've made all questions in FAQ files level 3 headings. Recommend leaving code ticks off and updated CD FAQ questions to be level 3 headings so they display in the mini toc. (Recently discussed in docs council meeting.

**Comment by brian-f-harness** [comment]
@SushrutHarness @aimurphy Are we okay to merge this? If we keep it open longer, it'll cause merge conflicts in multiple FAQ files.

**Review by brian-f-harness — DISMISSED** [review]

**Review by aimurphy — DISMISSED** [review]
approved from CI perspective

**Comment by aimurphy** [comment]
> @SushrutHarness @aimurphy Are we okay to merge this? If we keep it open longer, it'll cause merge conflicts in multiple FAQ files.

I am good from CI perspective

**Comment by brian-f-harness** [comment]
@douglas-j-bothwell Do you approve for STO?

**Comment by brian-f-harness** [comment]
> approved from CI perspective. Please wait for approval from other writers before merging.

@aimurphy Yes, thanks. I just realized STO was here too. I added a comment for Doug's approval.

**Review by brian-f-harness — APPROVED** [review]
Approved for PL/delegate

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 3]
```diff
@@ -6872,3 +6872,10 @@ Users can utilize the following API to retry a failed stage: [retryPipeline](htt
 #### When attempting to import a **.yaml file** from GitHub to create a new pipeline, the message `This file is already in use by another pipeline` is displayed. Given that there are no other pipelines in this project, is there a possibility of a duplicate entry that I may not be aware of?
 
 It's possible that there are two pipeline entities in the database, each linked to the same file path from the Harness account and the GitHub URL. Trying to import the file again may trigger the `File Already Imported` pop-up on the screen. However, users can choose to bypass this check by clicking the `Import` button again.
+
+#### Why is my GitOps Sync Step Failing with the error "Application does not correspond to the service(s) selected in the pipeline execution"?
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6876]:
```suggestion
#### Why is the GitOps Sync step failing with the error, "Application does not correspond to the service(s) selected in the pipeline execution"?
```

**Comment by douglas-j-bothwell** [comment]
@brian-f-harness @aimurphy I tweaked the STO FAQ a little. Will approve once the preview build finishes.

File: kb/continuous-delivery/continuous-delivery-faqs.md [hunk 4]
```diff
@@ -6872,3 +6872,10 @@ Users can utilize the following API to retry a failed stage: [retryPipeline](htt
...
+```
+Application{name: '$APP_NAME', agentIdentifier: '$AGENT_ID', errorMessage: 'Application does not correspond to the service(s) selected in the pipeline execution.'}
+```
+This error is caused by an inconsistency in the Deploy Stage's Service and the selected Application's Services. GitOps cannot sync a Service that isn't correlated to the Application. To fix this, navigate to the GitOps Application in the `GitOps` Menu under the `Continuous Delivery & GitOps` module and select `App Details`. Once there, confirm that the Service that's configured for the Application is the same as the Service configured in the Pipeline's Deploy Stage.
```
Comment (roshnisarangadharan) [inline kb/continuous-delivery/continuous-delivery-faqs.md:6880]:
```suggestion
This error is caused by an inconsistency in the deploy stage and the selected application's services. GitOps cannot sync a service that isn't correlated to the application. To fix this, go to the Continuous Delivery module, and select **GitOps** > **Applications** and then select the application. In **App Details**, check if the service that's configured for the application is the same as the service configured in the pipeline's deploy stage.
```

**Review by roshnisarangadharan — APPROVED** [review]
Updated the CD FAQ and approved.

**Review by brian-f-harness — APPROVED** [review]

**Review by douglas-j-bothwell — APPROVED** [review]
STO content approved.

**Pull request merged**